go.work

```diff
@@ -1,4 +1,4 @@
-go 1.21
+go 1.22
```

Comment:
This was recently discussed as having downstream effects: https://github.com/opentdf/platform/pull/1366#discussion_r1719940378

cc @pflynn-virtru and @dmihalcik-virtru

Reply:
I think we have no choice to force users to upgrade. We introduced a breaking change in our regex that is only supported in 1.22.

Reply:
I'm not extremely familiar with the `go.work` workspace setup intricacies, but it looks like it forces `1.22` if a module in the workspace requires it: 
![Screenshot 2024-08-20 at 3 31 32 PM](https://github.com/user-attachments/assets/f8d33e56-5c2e-44ec-a24a-25280081c194)

Unfortunately, the breaking change necessitates this or we get a panic on builds from `1.21` (see #1406).